src/cargo/ops/fix.rs

```diff
@@ -688,23 +688,19 @@ impl FixArgs {
         process::exit(1);
     }
 
-    fn prepare_for_edition_resolve(&self) -> Option<&str> {
-        match &self.prepare_for_edition {
+    fn prepare_for_edition_resolve(&self) -> Option<Edition> {
+        match self.prepare_for_edition {
             PrepareFor::Edition(s) => Some(s),
             PrepareFor::Next => Some(self.next_edition()),
             PrepareFor::None => None,
         }
     }
 
-    fn next_edition(&self) -> &str {
-        match self.enabled_edition.as_deref() {
-            // 2015 -> 2018,
-            None | Some("2015") => "2018",
-
-            // This'll probably be wrong in 2020, but that's future Cargo's
-            // problem. Eventually though we'll just add more editions here as
-            // necessary.
-            _ => "2018",
+    fn next_edition(&self) -> Edition {
+        match self.enabled_edition {
+            None | Some(Edition::Edition2015) => Edition::Edition2018,
+            Some(Edition::Edition2018) => Edition::Edition2021,
+            Some(Edition::Edition2021) => Edition::Edition2021,
```

Comment:
This matches the old behaviour of *2018 → 2018*. Should this give an error instead?